VOLA/VOLATests/SecretKeysManagerUnitTests.swift

```
@@ -0,0 +1,72 @@
+//
+//  SecretKeysManagerUnitTests.swift
+//  VOLA
+//
+//  Created by Connie Nguyen on 7/23/17.
+//  Copyright © 2017 Systers-Opensource. All rights reserved.
+//
+
+import XCTest
+
+class SecretKeysManagerUnitTests: XCTestCase {
+    /// Mock of SecretKeyManager class for testing purposes
+    class MockSecretKeyManager: SecretKeyManagerProtocol {
+        /// Name of plist file to load. Variable so valid and invalid files may be tested
+        var fileName: String = SecretKeysConstants.fileName
+
+        /**
+        Required to conform to SecretKeyManagerProtocol. Properly loading a dictionary from a plist and 
+         returning a default value if unable to load data. This is the subject of these unittests.
+        */
+        lazy var keys: [String: String] = {
+            guard let path = Bundle(for: SecretKeysManagerUnitTests.self).path(forResource: self.fileName, ofType: "plist"),
+                let keyDict = NSDictionary(contentsOfFile: path) as? [String: String] else {
+                    return [:]
+            }
+
+            return keyDict
+        }()
+
+        /// Stubbed function for retreiving value from stored plist
+        func value(for key: String) -> String {
+            guard let value = keys[key] else {
+                return ""
```

As a golden rule IMO, every unexpected behaviour should fail early on so we can catch that behaviour as soon as possible. In this case you are returning "" which can hide an potential error